extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -248,5 +251,212 @@ export class ProjectsController {
+				model.database = profile.databaseName;
+			}
+
+			// Get project name
+			let newProjName = await this.getProjectName(model.database);
+			if (!newProjName) {
+				throw new Error(constants.projectNameRequired);
+			}
+			model.projName = newProjName;
+
+			// Get extractTarget
+			let extractTarget: ExtractTarget = await this.getExtractTarget();
+			if (!extractTarget || extractTarget === -1) {
+				throw new Error(constants.extractTargetRequired);
+			}
+			model.extractTarget = extractTarget;
+
+			// Get folder location for project creation
+			let newProjUri = await this.getFolderLocation(model.extractTarget);
+			if (!newProjUri) {
+				throw new Error(constants.projectLocationRequired);
+			}
+
+			// Set project folder/file location
+			let newProjFolderUri;
+			if (extractTarget !== ExtractTarget.file) {
+				newProjFolderUri = newProjUri;
+			} else {
+				// Get folder info, if extractTarget = File
+				newProjFolderUri = Uri.file(path.dirname(newProjUri.fsPath));
+			}
+
+			//Check folder is empty
+			let isEmpty: boolean = await this.isDirEmpty(newProjFolderUri.fsPath);
+			if (!isEmpty) {
+				throw new Error(constants.projectLocationNotEmpty);
+			}
+			// TODO: what if the selected folder is outside the workspace?
+			model.filePath = newProjUri.fsPath;
+
+			//Set model version
+			model.version = '1.0.0.0';
+
+			// Call ExtractAPI in DacFx Service
+			await this.importApiCall(model);
+			// TODO: Check for success
+
+			// Create and open new project
+			const newProjFilePath = await this.createNewProject(newProjName as string, newProjFolderUri as Uri);
+			const project = await this.openProject(Uri.file(newProjFilePath));
+
+			//Create a list of all the files and directories to be added to project
+			let fileFolderList: string[] = await this.generateList(model.filePath);
+
+			// Add generated file structure to the project
+			await project.addToProject(fileFolderList);
+
+			//Refresh project to show the added files
+			this.refreshProjectsTree();
+		}
+		catch (err) {
+			this.apiWrapper.showErrorMessage(utils.getErrorMessage(err));
+		}
+	}
+
+	private async getProjectName(dbName: string): Promise<string | undefined> {
+		let projName = await this.apiWrapper.showInputBox({
+			prompt: constants.newDatabaseProjectName,
+			value: `DatabaseProject${dbName}`
+		});
+
+		projName = projName?.trim();
+
+		return projName;
+	}
+
+	private async getExtractTarget(): Promise<ExtractTarget> {
+		let extractTarget: ExtractTarget;
+
+		const extractTargetOptions: QuickPickItem[] = [];
+
+		let keys: string[] = Object.keys(ExtractTarget).filter(k => typeof ExtractTarget[k as any] === 'number');
+
+		keys.forEach(targetOption => {
+			let pascalCaseTargetOption: string = utils.toPascalCase(targetOption);	// for better readability
+			extractTargetOptions.push({ label: pascalCaseTargetOption });
+		});
+
+		let input = await this.apiWrapper.showQuickPick(extractTargetOptions.slice(1), {		//Ignore the first option to create Dacpac
```

Comment:
Can we do something like `extractTargetOptions.filter(x => x !== extractTargetOptions.Dacpac)` instead?  If those enum values ever get reordered, `slice(1)` will break.